### ISSUE TITLE
What factors do think about when you pick a repo and stick with it long term?

### ISSUE THREAD

**Issue by karuppiah7890** [karuppiah7890/ama#1]
Source of Q: https://twitter.com/ArasiyalaSagajm/status/1211734185511182336?s=20

**Comment by karuppiah7890** [comment]
I have only contributed to one repo for a long time - Helm and some of it's ecosystem. I can only say with that experience. 

To start with, I have never thought about long term when picking the repo. Previous to Helm I used to randomly check issues in repos and contribute. And then I chose Helm.

Some factors that I have keeping in mind while choosing are:
1. Why do I want to contribute to this? Do I know the why?
2. Is it written in a language that I know? - I have rarely chosen other languages. May be the next things I'll think about is - Can I at least manage a bit? Like learn it may be on the go? How much effort is that?
3. Is there anything that I can work on? Especially bugs. I could write a small blog on why bugs. 
4. Do I know the domain of the software, what it does? Or may be I can use it and understand a bit. If I cannot properly understand what it does from a user's perspective, it's gonna be very hard to contribute

Finally, for long term, I haven't thought of it before. I can say some things on it though. I think long term contribution is possible 
1. with patience, consistency
2. If there are more and more ways and possibilities to contribute to the project in the long term and to gain experience for the future contributions
3. If the person is interested in long term contributions and if they see benefits

**Comment by karuppiah7890** [comment]
I think there's also more to me choosing helm and it becoming a long term thing for me -

I was just exploring things. I chose it because I just knew about it through my day job. In my team we use Helm a lot. We also use multiple other tools and software. Like Kubernetes, Consul, Terraform and more! I just chose one, and other factors include - the language it was written in like I already said. Helm is written in golang and so are many projects that we use. And we also use golang predominantly, so it was a good opportunity to contribute and learn a bit too. Also, I usually pick and contribute to existing software (compared to creating my own) and I think I have been looking at the big ones - ones with lots of production users. It creates a high impact. After some contributing experience, I started liking it. Analyzing crazy issues. Fixing bugs. There was a lot of opportunity for contributions which made it long term for me. Especially a new version coming out - alpha releases, beta releases, release candidates, with lots of bugs that needed to be fixed
